pkg/cmd/issue/view/view_test.go

```diff
@@ -16,11 +16,37 @@ import (
 	"github.com/cli/cli/v2/pkg/cmdutil"
 	"github.com/cli/cli/v2/pkg/httpmock"
 	"github.com/cli/cli/v2/pkg/iostreams"
+	"github.com/cli/cli/v2/pkg/jsonfieldstest"
 	"github.com/cli/cli/v2/test"
 	"github.com/google/shlex"
 	"github.com/stretchr/testify/assert"
 )
 
+func TestJSONFields(t *testing.T) {
+	jsonfieldstest.ExpectCommandToHaveJSONFields(t, NewCmdView, []string{
+		"assignees",
+		"author",
+		"body",
+		"closed",
+		"comments",
+		"createdAt",
+		"closedAt",
+		"id",
+		"labels",
+		"milestone",
+		"number",
+		"projectCards",
+		"projectItems",
+		"reactionGroups",
+		"state",
+		"title",
+		"updatedAt",
+		"url",
+		"isPinned",
+		"stateReason",
+	})
+}
```

Comment:
Is it worthwhile to have a PR version of this test, ensuring none of the issue-only fields are emitted?

Reply:
I think if we like the pattern we should probably extend it to every command, though maybe do it in several tranches to let it cook for a bit. I think it'll probably be a pretty interesting exercise in auditing what we actually expose.

One very important thing is that the tests **must** declare the fields separately from the implementation. For example, we can't just use `api.IssueFields` in the test.

Reply:
I decided to apply this to `gh issue view` and `gh pr view` in this PR, and let's let it simmer for a bit. I want to get the bug fixed and not boil the ocean.